mh/database.py

```diff
@@ -497,6 +507,14 @@ def get_city(self, server_id, gate_id, index):
         assert 0 < index <= len(cities), "Invalid city index"
         return cities[index - 1]
 
+    def reserve_city(self, server_id, gate_id, index, reserve):
+        reserved_status = self.get_city(server_id, gate_id, index).reserved
+        if reserve is True and reserved_status is not None and\
+            time.time()-reserved_status < RESERVE_DC_TIMEOUT:
+            return False
+        self.get_city(server_id, gate_id, index).reserve(reserve)
+        return True
```

Comment:
When possible, let's avoid calling functions more than necessary. The following should be equivalent:
```python
def reserve_city(self, server_id, gate_id, index, reserve):
    city = self.get_city(server_id, gate_id, index)
    reserve_time = city.reserved
    if reserve and reserve_time and time.time()-reserve_time < RESERVE_DC_TIMEOUT:
        return False
    city.reserve(reserve)
    return True
```

Reply:
Fixed